src/tree/values/FormattingElement.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Represents phrasing content (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+ */
+class FormattingElement {
+	/**
+	 * Represents phrasing content (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+	 *
+	 * @param {string} tag the HTML-tag of this element (e.g. "strong", "a", etc.)
+	 * @param {number} start the start position of the tag within the text.
+	 * @param {number} end the end position of the tag within the text.
+	 * @param {Object} [attributes] the attributes (as key-value pairs, e.g. "href='...'" => { href: '...' } ).
+	 */
+	constructor( tag, start, end, attributes ) {
+		this.tag = tag;
```

Comment:
Should we type check these variables?

Reply:
In my opinion having the JSDoc types is enough of a barrier to prevent this function being called with the wrong parameters.